app/models/capture.js

```diff
@@ -61,41 +64,58 @@ class CaptureBoard {
     return new Promise((fulfill, reject) => {
       let subject = "Captured Conversation from New Contact";
       let deviceName = "New Captured Method";
-      
-      // Query 1: Update convo with CM and client
-      db("convos")
-        .where("convid", conversationId)
-        .andWhere("client", null)
-        .update({
-          subject: subject,
-          client: client,
-          cm: user,
-          accepted: true,
-          open: true
-        })
-      .then((success) => {
-
-      // Query 2: Close all other conversation that client has open
-      db("convos")
-        .whereNot("convid", conversationId)
-        .andWhere("cm", user)
-        .andWhere("client", client)
-        .update({ open: false })
-      .then((success) => {
-
-      // Query 3: Gather comm forms used in that conversation (should be only one)
-      db("msgs")
-        .where("convo", conversationId)
-        .andWhere("inbound", true)
-        .pluck("comm")
-      .then((comms) => {
-
-      // Query 4: Gather comms that the CM currently has
-      db("commconns")
-        .where("client", client)
-        .andWhere("retired", null)
-        .pluck("comm")
-      .then((clientcomms) => {
+      let clientcomms, comms, conversation;
+
+      Conversations.findById(conversationId)
+      .then((resp) => {
+        conversation = resp;
+        return Messages.findByConversation(conversation)
+      }).then((resp) => {
+        let messages = resp;
+        let commId = messages[0].comm;
+        return Communications.findById(commId)
+      }).then((resp) => {
+        let communcation = resp;
+        return Conversations.closeAllCapturedWithCertainCommunication(communcation)
+      }).then((resp) => {
+        // Query 1: Update convo with CM and client
+        return db("convos")
+          .where("convid", conversationId)
+          .update({
+            subject: subject,
+            client: client,
+            cm: user,
+            accepted: true,
+            open: true
+          })
+          .returning("*");
+      }).then((resp) => {
+        return Conversations.findById(conversationId)
+      }).then((resp) => {
+
+        // Query 2: Close all other conversation that client has open
+        return db("convos")
```

Comment:
roger, will do. `associateConversation` is a sort of mess and im just leaving it for another day to clean up